### PR TITLE
Adding docs for two loss functions

### PR DESCRIPTION
**Pull request by hsezhiyan**
Docs for mean_absolute_error.py and mean_squared_error.py.

This is a resend of #6235 (which didn't follow the exact styling guideline). Hopefully this one has the correct styling.

### CODE REVIEW

File: chainer/functions/loss/mean_absolute_error.py [hunk 1]
```diff
@@ -41,8 +41,11 @@ def backward(self, indexes, grad_outputs):
 def mean_absolute_error(x0, x1):
     """Mean absolute error function.
 
-    This function computes mean absolute error between two variables. The mean
-    is taken over the minibatch.
+    This function computes the mean absolute error between two variables. The
+    mean is taken over the minibatch. Args ``x0`` and ``x1`` must have the
+    dimensions. This function first calculates the absolute value differences
```
Comment (Crissman) [inline chainer/functions/loss/mean_absolute_error.py]:
```suggestion
    same dimensions. This function first calculates the absolute value of the differences
```

File: chainer/functions/loss/mean_absolute_error.py [hunk 2]
```diff
@@ -41,8 +41,11 @@ def backward(self, indexes, grad_outputs):
...
+    of corresponding elements in x0 and x1, and then returns the mean of those
```
Comment (Crissman) [inline chainer/functions/loss/mean_absolute_error.py]:
```suggestion
    between the corresponding elements in x0 and x1, and then returns the mean of those
```

**Comment by Crissman** [comment]
Take a look at the `make html` warnings: /Users/crissman/anaconda3/lib/python3.7/site-packages/chainer-6.0.0b2-py3.7-macosx-10.7-x86_64.egg/chainer/functions/loss/mean_absolute_error.py:docstring of chainer.functions.mean_absolute_error:34: WARNING: Bullet list ends without a blank line; unexpected unindent.
/Users/crissman/anaconda3/lib/python3.7/site-packages/chainer-6.0.0b2-py3.7-macosx-10.7-x86_64.egg/chainer/functions/loss/mean_absolute_error.py:docstring of chainer.functions.mean_absolute_error:42: WARNING: Bullet list ends without a blank line; unexpected unindent.
/Users/crissman/anaconda3/lib/python3.7/site-packages/chainer-6.0.0b2-py3.7-macosx-10.7-x86_64.egg/chainer/functions/loss/mean_squared_error.py:docstring of chainer.functions.mean_squared_error:32: WARNING: Bullet list ends without a blank line; unexpected unindent. 
/Users/crissman/anaconda3/lib/python3.7/site-packages/chainer-6.0.0b2-py3.7-macosx-10.7-x86_64.egg/chainer/functions/loss/mean_squared_error.py:docstring of chainer.functions.mean_squared_error:40: WARNING: Bullet list ends without a blank line; unexpected unindent.

It's a good idea to run `cd docs; make clean; SPHINXOPTS= make html` from the chainer directory to check for errors, so you don't have to wait for the Travis CI test.

**Comment by Crissman** [comment]
@hsezhiyan Did you have a chance to look at those html warnings?

**Comment by Crissman** [comment]
Closer. Down to only one HTML error. Please take a look.

**Comment by Crissman** [comment]
@hsezhiyan For Flake8, indentation cannot contain tab characters. Please replace the tabs with spaces, and then use `flake8` to confirm fixed.

Thanks!

**Comment by hsezhiyan** [comment]
@Crissman changed tabs to spaces.

File: chainer/functions/loss/mean_absolute_error.py [hunk 3]
```diff
@@ -53,5 +56,36 @@ def mean_absolute_error(x0, x1):
...
+        >>> F.mean_absolute_error(x, y)
+        variable(2.)
+        >>> x = np.array([1, 2, 3, 4, 5, 6]).astype(np.float32)
+        >>> y = np.array([7, 8, 9, 10, 11, 12]).astype(np.float32)
+        >>> F.mean_absolute_error(x, y)
+        variable(6.)
+
+        2D array example:
+
+        - In this example, there are 4 elements, and thus 4 errors
```
Comment (Crissman) [inline chainer/functions/loss/mean_absolute_error.py]:
```suggestion
        In this example, there are 4 elements, and thus 4 errors
```

File: chainer/functions/loss/mean_absolute_error.py [hunk 4]
```diff
@@ -53,5 +56,36 @@ def mean_absolute_error(x0, x1):
...
+        >>> x = np.array([[1, 2], [3, 4]]).astype(np.float32)
+        >>> y = np.array([[8, 8], [8, 8]]).astype(np.float32)
+        >>> F.mean_absolute_error(x, y)
+        variable(5.5)
+
+        3D array example:
+
+        - In this example, there are 8 elements, and thus 8 errors
```
Comment (Crissman) [inline chainer/functions/loss/mean_absolute_error.py]:
```suggestion
        In this example, there are 8 elements, and thus 8 errors
```

File: chainer/functions/loss/mean_squared_error.py [hunk 5]
```diff
@@ -55,5 +56,36 @@ def mean_squared_error(x0, x1):
...
+        >>> F.mean_squared_error(x, y)
+        variable(7.5)
+        >>> x = np.array([1, 2, 3, 4, 5, 6]).astype(np.float32)
+        >>> y = np.array([7, 8, 9, 10, 11, 12]).astype(np.float32)
+        >>> F.mean_squared_error(x, y)
+        variable(36.)
+
+        2D array example:
+
+        - In this example, there are 4 elements, and thus 4 errors
```
Comment (Crissman) [inline chainer/functions/loss/mean_squared_error.py]:
```suggestion
        In this example, there are 4 elements, and thus 4 errors
```

File: chainer/functions/loss/mean_squared_error.py [hunk 6]
```diff
@@ -55,5 +56,36 @@ def mean_squared_error(x0, x1):
...
+        >>> x = np.array([[1, 2], [3, 4]]).astype(np.float32)
+        >>> y = np.array([[8, 8], [8, 8]]).astype(np.float32)
+        >>> F.mean_squared_error(x, y)
+        variable(31.5)
+
+        3D array example:
+
+        - In this example, there are 8 elements, and thus 8 errors
```
Comment (Crissman) [inline chainer/functions/loss/mean_squared_error.py]:
```suggestion
        In this example, there are 8 elements, and thus 8 errors
```

**Comment by Crissman** [comment]
@hsezhiyan Please update to fix the HTML errors, so we can proceed, thanks!

**Comment by Crissman** [comment]
Jenkins, test this please.

**Comment by Crissman** [comment]
LGTM. Merging.

**Pull request merged**